devnet-sdk/devstack/sysgo/sync_test.go

```diff
@@ -116,3 +116,106 @@ func TestL2CLResync(gt *testing.T) {
 		// supervisor successfully connected with managed L2CLs
 	}
 }
+
+// TestL2CLSyncP2P checks that unsafe head is propagated from sequencer to verifier.
+// Tests started/restarted L2CL advances unsafe head via P2P connection.
+func TestL2CLSyncP2P(gt *testing.T) {
```

Comment:
This makes sense to add here for now, but maybe we should just make it a generic acceptance-test later, where it can also run against the kurtosis backend.

Just need a few changes to the `stack.L2CLNode` to expose the `apis.P2PClient` (see other comment) and make `shim.L2CLNode` have a helper method to acquire the interface similar to the rollup-client interface, by wrapping the underlying RPC with the bindings.

Reply:
Added at https://github.com/ethereum-optimism/optimism/pull/15462/commits/5ad42025340e333ac1cd671fd054a42904288e16